app/src/main/java/com/android/sample/ui/createRecipe/RecipeNameScreen.kt

```diff
@@ -0,0 +1,147 @@
+package com.android.sample.ui.createRecipe
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.android.sample.R
+import com.android.sample.model.recipe.CreateRecipeViewModel
+import com.android.sample.resources.C.Tag.RECIPE_NAME_BASE_PADDING
+import com.android.sample.resources.C.Tag.RECIPE_NAME_BUTTON_HEIGHT
+import com.android.sample.resources.C.Tag.RECIPE_NAME_BUTTON_WIDTH
+import com.android.sample.resources.C.Tag.RECIPE_NAME_FIELD_HEIGHT
+import com.android.sample.resources.C.Tag.RECIPE_NAME_FIELD_SPACING
+import com.android.sample.resources.C.Tag.RECIPE_NAME_FONT_SPACING
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Screen
+import com.android.sample.ui.theme.*
+
```

Comment:
For this is screen, i struggled a bit to use weight for everything. But i did as much as i could, if you still spot area of improvement let me know !! The ChefImage hasnt been added because i have struggle to make the whole thing responsive with it. Maybe we should meet to talk about how we can solve this

Reply:
Looks great to me, well done!